### PR TITLE
Doc updates for federation and link fixes

### PR DESCRIPTION
**Pull request by artberger**
# Description

* Removes demo environment federation docs, since the demo is not maintained
* Updates the link for external auth to v3

# Context

Discussions in Slack:
* Federation: https://solo-io-corp.slack.com/archives/CEDCS8TAP/p1635530712054300
* Link: https://solo-io-corp.slack.com/archives/CHJV572TG/p1635521367031200

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- N/A If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- N/A I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- N/A I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: docs/content/guides/gloo_federation/getting_started.md [hunk 1]
```diff
@@ -1,259 +0,0 @@
----
-title: Getting Started
```
Comment (jenshu) [inline docs/content/guides/gloo_federation/getting_started.md:2]: will this page be replaced by a new guide? or do we already have it all covered on another page?

Comment (artberger) [reply docs/content/guides/gloo_federation/getting_started.md:2]: So for now the users would just have to follow the more specific sub pages, vs. having a nice demo getting started guide.

image

File: docs/content/guides/gloo_federation/service_failover.md [hunk 2]
```diff
@@ -10,20 +10,32 @@ When an Upstream fails or becomes unhealthy, Gloo Edge Federation can automatica
...
+Gloo Edge Enterprise version `1.5.0-beta4` or later is required to use failover. You can enable failover by setting following Helm value: `gatewayProxies.NAME.failover.enabled=true`.
+
+An example Helm override file for installing Gloo Edge with failover is:
+```yaml
+gatewayProxies:
+  gatewayProxy:
+    failover:
+      enabled: true
+    service:
+      type: NodePort
```
Comment (jenshu) [inline docs/content/guides/gloo_federation/service_failover.md:32]: for enterprise, this all needs to be nested under a `gloo` field, right?

Comment (artberger) [reply docs/content/guides/gloo_federation/service_failover.md:32]: I just copied over what we had from the [Getting Started section](https://docs.solo.io/gloo-edge/latest/guides/gloo_federation/getting_started/#upgrading-gloo-edge-to-use-failover).

Maybe that was using OSS, so I will update it to enterprise here.

**Review by jenshu — DISMISSED** [review]

**Review by Rachael-Graham — APPROVED** [review]

**Pull request merged**